concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V1/Iterator.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-| This module tests calling a contract which creates an Iterator and ... todo
```

Comment:
todo

Can you add, related to the previous comment, that the test/checks are performed in the contract itself, meaning if a contract completes successfully that is deemed a passing test, and that no further checks (like checking whether the state is a specific value) are done in this file.